modin/experimental/engines/omnisci_on_native/frame/partition.py

```diff
@@ -119,13 +101,26 @@ def put(cls, obj):
 
         Returns
         -------
-        OmnisciOnRayFramePartition
+        OmnisciOnNativeFramePartition
             The new partition.
         """
-        return OmnisciOnRayFramePartition(
-            object_id=ray.put(obj), length=len(obj.index), width=len(obj.columns)
+        return OmnisciOnNativeFramePartition(
+            pandas_df=obj, length=len(obj.index), width=len(obj.columns)
         )
 
+    def wait(self):
+        """
+        Wait until the partition data is ready for use.
+
+        Returns
+        -------
+        pandas.DataFrame
+            The partition that is ready to be used.
+        """
+        if self.arrow_table is not None:
+            return self.arrow_table.to_pandas()
```

Comment:
why we do not call `.to_pandas()` in `get()` but call it here?

Reply:
`wait` method for OmniSci is mostly for compatibility purposes since we wait computations for OmniSci using `_execute` method in ASV benchmarks. But yes, I think we can drop `.to_pandas` call here.